pkg/controller/import-controller.go

```diff
@@ -74,6 +95,56 @@ func (ic *ImportController) findImportPodFromCache(pvc *v1.PersistentVolumeClaim
 	return podList[0], nil
 }
 
+func (ic *ImportController) importerRequiresScratchSpace(pvc *v1.PersistentVolumeClaim) bool {
+	contentType := getContentType(pvc)
+	// All archive requires scratch space.
+	if contentType == "archive" {
+		glog.V(3).Infof("Archive content type, requires scratch space\n")
+		return true
+	}
+	source := getSource(pvc)
+	// Invalid Sources will have gotten rejected by the web hook, so this is fine.
+	if source != SourceHTTP && source != SourceNone {
```

Comment:
So, what about the switch case? Without it - it is easy to avoid explicit consideration regarding need for scratch space in particular import and always ask for scratch PVC

Reply:
With a switch we explicitly note which ones require scratch space vs which ones don't, so adding a new source type requires altering this switch if it requires scratch space vs. this which is the reverse, if the new one doesn't require scratch space we have to alter the if statement. Its a matter of preference I guess, and I have no objection to either, so I will put in the switch.